File: raspberry/SmartHome/arduinodevices/src/main/java/smarthome/raspberry/arduinodevices/ArduinoDeviceAPI.java
//package smarthome.raspberry.arduinodevices;
//
//import retrofit2.Call;
//import retrofit2.http.GET;
//import retrofit2.http.POST;
//import retrofit2.http.Query;
//
//public interface ArduinoDeviceAPI {
//
//    @GET("service")
//    Call<ArduinoControllerResponse> controllerReadRequest(@Query("index") int serviceIndex);
//
//    @POST("service")
//    Call<ArduinoControllerResponse> controllerWriteRequest(
//            @Query("index") int serviceIndex, @Query("value") String value);
//
//}

File: raspberry/arduinodevices/src/main/java/smarthome/raspberry/arduinodevices/ArduinoDeviceApi.kt
package smarthome.raspberry.arduinodevices

import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface ArduinoDeviceApi {

    @GET("service")
    suspend fun controllerReadRequest(@Query("index") serviceIndex: Int): ArduinoControllerResponse

    @POST("service")
    suspend fun controllerWriteRequest(
            @Query("index") serviceIndex: Int,
            @Query("value") value: String): ArduinoControllerResponse

}


class ArduinoControllerResponse(val response: String)

